pandas/core/arrays/base.py

```diff
@@ -653,15 +661,59 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __unicode__(self):
+        from pandas.io.formats.printing import format_object_summary
```

Comment:
Can you add a comment here explaining that is basically defining `__repr__` but with some py2/3 compat things handled for you? Because EA authors might wonder why this is not `__repr__`

Reply:
I didn’t really realize the effects of this on downstream users of EA. This really should be implemented in repr, so that we don’t interact poorly with a custom str method.